pre-commit.sh.tpl

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+RED='\033[0;31m'
+GREEN='\033[0;33m'
+CYAN='\033[0;36m'
+NC='\033[0m' # No Color
+
+function check {
+    name=$1
+    prefix=$2
+    files=$(git diff --cached --name-only --diff-filter=ACM | grep $prefix)
+    shift; shift;
+    echo -n "checking $name ";
+    if [ -z "$files" ]; then 
+        echo -e "${CYAN}[NOT APPLICABLE]${NC}"
+        exit 0;
+    fi
+
+    cd $(git rev-parse --show-toplevel)
+    
+    out=$($@ $files)
```

Comment:
In an unlikely event that we have too many files, we might need `xargs` here. However, I wouldn't bother at this time.

Reply:
I think it is highly unlikely, typically commit touches only a few files so this wouldn't be a problem